### ISSUE TITLE
SV OnLuaError for client errors

### ISSUE THREAD

**Issue by wrefgtzweve** [Facepunch/garrysmod-requests#2371]
### Details
https://wiki.facepunch.com/gmod/GM:OnLuaError doesn't currently pass CL errors on SV
Having some kind of OnClientLuaError on SV would be incredibly useful for custom error logging tools, the errors are already being networked and ratelimited so it should be pretty safe to add.
It understandably wouldnt have the full stack like GM:OnLuaError but that's fine, just having the stacktrace like regular would be great.

**Comment by TeddiO** [comment]
Native support would be neat, but https://github.com/danielga/gm_luaerror might help you out.

**Comment by Astralcircle** [comment]
I think this will be a great addition considering that client errors are already sent to the server console

**Comment by JarosLucky** [comment]
Related to HandleClientLuaError
https://github.com/RaphaelIT7/gm_luaerror/blob/a9f4d3cc29ae1574deaf960fadb0e4876d32c67a/source/server/server.cpp#L102-L122

**Comment by JarosLucky** [comment]
> Added `GM:OnClientLuaError( strErr, ply )`.
Can we have args like `GM:OnLuaErrror`?
Sth like `GM:OnClientLuaError( string error, string realm, table stack, string name, string id )`. Player branch would be nice, but I don't know if it's possible.

**Comment by JarosLucky** [comment]
> Added `GM:OnClientLuaError( strErr, ply )`.
Can we have args like `GM:OnLuaErrror`?
Sth like `GM:OnClientLuaError( string error, string realm, table stack, string name, string id )`. Player branch would be nice, but I don't know if it's possible.

**Comment by robotboy655** [comment]
Not without a network breaking update. The only thing you don't have access to would be the addon ID, the rest should be available from the error string, the addon name, the stack, and the realm would be always `client`.

**Comment by robotboy655** [comment]
They parse the string just like you can.

https://github.com/danielga/gm_luaerror/blob/7c7046dd3ea35d7df4fc8c7c9b4c042f399bdce3/source/common/common.cpp#L52

I guess the official hook can as well.

**Comment by JarosLucky** [comment]
> I guess the official hook can as well.
Yeah, I'm thinking about that too. At least it would make the official hook more like a server hook, which would definitely be easier to use. Also, thanks for the quick response.

**Comment by JarosLucky** [comment]
> I guess the official hook can as well.
Yeah, I'm thinking about that too. At least it would make the official hook more like a server hook, which would definitely be easier to use. Also, thanks for the quick response.

**Comment by robotboy655** [comment]
I have made some changes to have the arguments be ( error, ply, stack, addon_name ), but it will likely not be reliable with weird addon/file names.

**Comment by JarosLucky** [comment]
> Not without a network breaking update.

To be honest I don't think there is a need to change the networking to become more “insignificant information” here. The only thing that might be useful is to somehow have information about what branch a player is on. And that can already be done independently if necessary. The same CFC Err Forwarder gets this information itself. Thanks for greatly simplifying the use and no longer needing a separate module to get client errors. All in all, thanks for your work.
